### ISSUE TITLE
duplicate extensions in asdf 2.14

### ISSUE THREAD

**Issue by braingram** [asdf-format/asdf#1254]
In certain circumstances, weldx.asdf.utils.get_weldx_extension fails due to duplicate extensions: https://github.com/BAMWelDX/weldx/issues/825

Installing a local clone of the weldx repository and running the following while in the repository parent directory produces the error:
```python
import asdf
import weldx

af = asdf.AsdfFile()
ctx = af._create_serialization_context()
weldx.asdf.util.get_weldx_extension(ctx)
```

```python
  File "/Users/bgraham/projects/221201_weldx_downstream_test/weldx/get_extension.py", line 6, in <module>
    weldx.asdf.util.get_weldx_extension(ctx)
  File "/Users/bgraham/projects/221201_weldx_downstream_test/weldx/weldx/asdf/util.py", line 459, in get_weldx_extension
    raise ValueError("Could not determine correct weldx extension.")
ValueError: Could not determine correct weldx extension.
```

Running the same code outside the weldx repository does not result in an error. There appears to be an issue with importlib:
https://github.com/pypa/setuptools/issues/3649

Forcing the use of importlib-metadata==4.11.4 (installed via pip, tested with python 3.10.6) by modifying these lines fixes the issue: https://github.com/asdf-format/asdf/blob/ad6697147366bc77b5d9cf3c62871324bc1db20a/asdf/entry_points.py#L4-L7

It is unclear to me if this is intended behavior of importlib or a bug that per this comment appears to be in the python stdlib https://github.com/pypa/setuptools/issues/3649#issuecomment-1294229882

I'm not sure if other extensions are being duplicated or if this duplication is causing any issues (aside from the one in weldx).

One possible fix would be to put a lower pin on importlib-metadata and to not use the stdlib importlib for python 3.10/3.11.

**Comment by marscher** [comment]
Thanks! This precisely describes the problem and the workaround of only relying on one dependency (importlib_metadata) seems like a good approach.

I think the duplication of the extensions is not a problem per se. It would be hard to tell which extension is used then (there is some sorting going on somewhere, but you cannot be sure, which version you'll get). In weldx there is an explicit check, that we only receive one (matching) extension, so the underlying "problem" actually appears.

**Comment by braingram** [comment]
Fixed in #1260
